src/v/cluster/rm_stm.cc

```diff
@@ -1814,13 +1806,66 @@ ss::future<raft::stm_snapshot> rm_stm::do_take_local_snapshot(
     }
     auto snapshot_offset = last_applied_offset();
     apply_units.return_all();
+    auto write_lock_units = co_await _state_lock.hold_write_lock();
```

Comment:
Do we need to hold this lock while we persist the abort snapshots? After all we modify `aborted` and `abort_indexes` in a single continuation and don't require them to be stable while we write the snapshots.